### PR TITLE
Comms console now adds slurring etc, checks if user can talk

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
A few things already indicate that these are indeed vocal, this cements that. Adds a speech check and runs the message through treat_message to add slurring, unintelligible etc.

## Why It's Good For The Game
More consistent. Someone really should add languages to these.

## Changelog
:cl: Skoglol
tweak: Comms console announcement now has a speech check and adds slurring etc.
/:cl:

### CODE REVIEW

**Comment by Tharcoonvagh** [comment]
Finally

**Comment by trollbreeder** [comment]
finally now i can announce xenobiology WGW over the radio while drunk and stammering at the same time

**Comment by itseasytosee** [comment]
I was always under the impression that these were typed messages IC because this did not exist and mute people could do it but it would make sense if we fixed that

**Comment by Neo-0** [comment]
makes no sense since you need to type a WORD, not speak to make an announce on a CONSOLE

**Comment by kriskog** [comment]
What are microphones

**Comment by HomocidlePancakes** [comment]
In my canon it’s a TTS machine.

**Comment by kriskog** [comment]
Go away mime main

**Comment by Neo-0** [comment]
we don't have microphones, you type in the console, isn't?

**Comment by trollbreeder** [comment]
> we don't have microphones, you type in the console, isn't?

WHAT IS AN INTERCOM?! WHaTS A RADIO HeADSET?! WHAT IS A STATION BOUNCED RADIO?!?!!7!!1!?!! /s

**Comment by Neo-0** [comment]
> > we don't have microphones, you type in the console, isn't?
> 
> WHAT IS AN INTERCOM?! WHaTS A RADIO HeADSET?! WHAT IS A STATION BOUNCED RADIO?!?!!7!!1!?!! /s

it's about the 'make announce' in the comms console, not telecomms itself

**Comment by trollbreeder** [comment]
next we're gonna tie announcements to the z level of the comms console

**Comment by cacogen** [comment]
nice. saw a well-intentioned mime use it last night and of course had to decapitate her. also i wish you couldn't hear these on other z-levels (except mining, i suppose). really kills the isolated charliestation vibe

**Comment by itseasytosee** [comment]
I'm sorry I don't speak non-cannon.

**Comment by itseasytosee** [comment]
I'm sorry, I forgot that it makes you take a breathalyzer before making an announcement and automatically adjusts what you type

**Comment by XDTM** [comment]
Maybe it could also apply other text modifications, like the robotic span when the AI does it

**Comment by ATH1909** [comment]
Eh, I personally like being able to make an announcement to everyone as a mime without needing to call the shuttle to do so (shuttle calling doesn't require you to not be mute, right?).

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -698,6 +698,10 @@
 	var/input = stripped_input(user, "Please choose a message to announce to the station crew.", "What?")
 	if(!input || !user.canUseTopic(src, !issilicon(usr)))
 		return
+	if(!(user.can_speak())) //No more cheating, mime/random mute guy!
+		input = "..."
```
Comment (Cyberboss) [inline code/game/machinery/computer/communications.dm]: Should inform the user that they are mute in this case.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/communications.dm (hunk 1 above)
Comment (kriskog) [reply code/game/machinery/computer/communications.dm]: Done

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**